doc/doxygen/scripts/filter

```diff
@@ -35,12 +35,23 @@ while (<>)
     # We don't let doxygen put everything into a namespace
     # dealii. consequently, doxygen can't link references that contain an
     # explicit dealii:: qualification. remove it and replace it by the
-    # global scope
+    # global scope.
+    #
+    # Try to detect whether or not ::dealii:: is part of a macro (e.g., Assert):
+    # if it is then replace it with spaces to keep alignment correct.
```

Comment:
It's not actually a macro line you test (I had to scratch my head for a long time about this) but in fact about lines with a continuation-backslash at the end. Maybe describe this in the comment?